src/tools/rust-analyzer/crates/proc-macro-srv/src/dylib.rs

```diff
@@ -28,11 +28,14 @@ fn load_library(file: &Utf8Path) -> Result<Library, libloading::Error> {
 
 #[cfg(unix)]
 fn load_library(file: &Utf8Path) -> Result<Library, libloading::Error> {
+    // not defined by POSIX, different values on mips vs other targets
+    #[cfg(target_env = "gnu")]
+    use libc::RTLD_DEEPBIND;
     use libloading::os::unix::Library as UnixLibrary;
-    use std::os::raw::c_int;
-
-    const RTLD_NOW: c_int = 0x00002;
-    const RTLD_DEEPBIND: c_int = 0x00008;
+    // defined by POSIX
+    use libloading::os::unix::RTLD_NOW;
+    #[cfg(not(target_env = "gnu"))]
+    const RTLD_DEEPBIND: std::os::raw::c_int = 0x00008;
```

Comment:
This should be 0 for all targets where `RTLD_DEEPBIND` isn't supported.

Reply:
I didn't want to deviate for targets where I lack knowledge, or require syncing this up with `libc` (the crate) or various libc implementations. but if that is preferred I can do that of course :)